### ISSUE TITLE
Allow calling tf2onnx.convert programmatically

### ISSUE THREAD

**Issue by letmaik** [onnx/tensorflow-onnx#669]
Currently there is no straight-forward way to invoke tf2onnx.convert from Python. Adding this would make integration of tf2onnx into bigger pipelines easier.

**Comment by guschmue** [comment]
Does this work:
https://github.com/onnx/tensorflow-onnx#using-the-python-api
https://github.com/onnx/tensorflow-onnx/blob/master/examples/call_coverter_via_python.py

**Comment by letmaik** [comment]
That doesn't work. I think it behaves differently from the command line. First, it outputs a bunch of `Cannot infer shape for ...` warnings (which are irrelevant I think), but then it fails with:
```
...
  File "/home/.../miniconda3/lib/python3.7/site-packages/tf2onnx/tfonnx.py", line 708, in process_tf_graph
    onnx_nodes, op_cnt, attr_cnt, output_shapes, dtypes = tensorflow_to_onnx(tf_graph, shape_override)
  File "/home/.../miniconda3/lib/python3.7/site-packages/tf2onnx/tfonnx.py", line 123, in tensorflow_to_onnx
    return tflist_to_onnx(graph.get_operations(), shape_override)
  File "/home/.../miniconda3/lib/python3.7/site-packages/tf2onnx/tfonnx.py", line 67, in tflist_to_onnx
    dtypes[out.name] = utils.map_tf_dtype(out.dtype)
  File "/home/.../miniconda3/lib/python3.7/site-packages/tf2onnx/utils.py", line 190, in map_tf_dtype
    dtype = TF_TO_ONNX_DTYPE[dtype]
KeyError: tf.float32_ref
```

**Comment by guschmue** [comment]
tf2onnx expects a frozen graph and ```tf.float32_ref``` hints that there are Variables in the graph. 
You could do something like this:
https://github.com/onnx/tensorflow-onnx/blob/master/tf2onnx/loader.py#L24
or call freeze_session() directly. We did not intend to make this part of the official api but maybe we should.

**Comment by letmaik** [comment]
The README says "tf2onnx supports saved_model, checkpoint or frozen graph formats." ?

**Comment by guschmue** [comment]
yes, for the command line. For the python api I think most people to prefer to freeze in the caller in case there are corner cases the model wants to handle.
